dff/utils/testing/telegram.py

```diff
@@ -0,0 +1,182 @@
+from dff.script.core.message import Message, Attachments, Attachment, Button, Location
+from dff.messengers.telegram.interface import PollingTelegramInterface
+from dff.messengers.telegram.message import TelegramMessage, TelegramUI, RemoveKeyboard, _ClickButton
+
+
+async def get_bot_user(client: TelegramClient, username: str):
+    async with client:
+        return await client.get_entity(username)
+
+
+class TelegramTesting:
+    """Defines functions for testing."""
+
+    def __init__(self, pipeline: Pipeline, client: Optional[TelegramClient] = None, bot: Optional[User] = None):
+        if client is None:
+            tg_id = getenv("TG_API_ID")
+            tg_hash = getenv("TG_API_HASH")
+            if tg_id is not None and tg_hash is not None:
+                client = TelegramClient("anon", int(tg_id), tg_hash)
+            else:
+                raise RuntimeError("Telegram API credentials are not set, client argument is not specified")
+        self.client = client
+        """Telegram client (not bot). Needed to verify bot replies."""
+        self.pipeline = pipeline
+        if bot is None:
+            bot = getenv("TG_BOT_USERNAME")
+            if bot is None:
+                raise RuntimeError("Bot username is not set, bot argument is not specified")
+            bot = asyncio.run(get_bot_user(self.client, bot))
+        self.bot = bot
+        """Bot user (to know to whom to send messages from client)."""
+
+    async def send_message(self, message: Message, last_bot_messages: List[TlMessage]):
+        """Send a message from client to bot."""
+        if message.commands is not None:
+            if len(message.commands) != 1:
+                raise RuntimeError(f"Multiple commands are not used in telegram: {message.commands}")
+            command = message.commands[0]
+            if not isinstance(command, _ClickButton):
+                raise RuntimeError(f"Only `_ClickButton` command is supported by telegram: {command}")
+            button_clicked = False
+            for bot_message in last_bot_messages:
+                if bot_message.buttons is not None:
+                    if button_clicked:
+                        raise RuntimeError("Found multiple messages with buttons")
+                    await bot_message.click(i=command.button_index)
+                    return None
+        if message.attachments is None or len(message.attachments.files) == 0:
+            return await self.client.send_message(self.bot, message.text)
+        else:
+            if len(message.attachments.files) == 1:
+                attachment = message.attachments.files[0]
+                files = attachment.source
+            else:
+                files = [file.source for file in message.attachments.files]
+            return await self.client.send_file(self.bot, files, caption=message.text)
+
+    @staticmethod
+    async def parse_responses(responses: List[TlMessage], tmp_dir) -> Message:
```

Comment:
`tmp_dir` is so bad naming, and it should be better to make `tmp_dir` as optional, just create a dir in `/tmp` for this case.

Pls, change the   `tmp_dir`  for all functions

Reply:
https://github.com/deeppavlov/dialog_flow_framework/pull/13/commits/cbff5c9134d160be9e0307464225236b40a1a0e6